src/components/pages/lessons/overlay/go-pro-cta-overlay.tsx

```diff
@@ -2,30 +2,170 @@ import React, {FunctionComponent} from 'react'
+      })
+      await track('checkout: redirect to stripe', {
+        priceId,
+      })
+
+      stripeCheckoutRedirect({
+        priceId,
+        email: emailForCheckout,
+        stripeCustomerId: account?.stripe_customer_id,
+        authToken,
+        quantity,
+        coupon: state.context.couponToApply?.couponCode,
+      })
+    } else {
+      // we don't have a valid email for the checkout
+      await track('checkout: unable to proceed, no valid email', {
+        email: emailForCheckout,
+      })
+    }
+  }
+
   return (
     <div className="flex flex-col items-center p-4">
-      <h2 className="sm:text-2xl text-xl leading-tighter tracking-tight font-light text-center">
-        This lesson is an{' '}
-        <strong className="font-bold">egghead member exclusive</strong>.
+      <h2 className="sm:text-lg text-sm uppercase leading-tighter tracking-tight font-light text-center">
+        This lesson is for members only
       </h2>
-      <Link href="/pricing">
-        <a
-          onClick={() => {
-            track('clicked join cta on blocked lesson', {
-              lesson: lesson.slug,
-            })
+      <h1 className="sm:text-2xl text-xl leading-tighter tracking-tight font-light text-center">
+        Ready to take your career to the next level?
+      </h1>
+      <PlanPrice pricesLoading={pricesLoading} plan={currentPlan} />
+      {isEmpty(viewer) && (
+        <input
+          id="email"
+          type="email"
+          value={emailForCheckout}
+          onChange={(e) => {
+            setEmailForCheckout(e.target.value)
           }}
-          className="mt-8 text-white py-3 px-5 rounded-md text-base font-semibold bg-blue-600 hover:bg-indigo-600 hover:shadow-xl transition-all ease-in-out duration-300 hover:scale-105"
-        >
-          Become a Member
-        </a>
-      </Link>
+          placeholder="you@company.com"
+          className="py-3 text-gray-900 placeholder-gray-400 focus:ring-indigo-500 focus:border-blue-500 block w-full pl-10 border-gray-300 rounded-md"
+          required
+        />
+      )}
+      <button
+        className="w-full px-5 py-4 mt-8 font-semibold text-center text-white transition-all duration-300 ease-in-out bg-blue-600 rounded-md hover:bg-blue-700 hover:scale-105"
+        onClick={(_event) => {
+          track('clicked join cta on blocked lesson', {
+            lesson: lesson.slug,
+          })
+          send({type: 'CONFIRM_PRICE', onClickCheckout})
+        }}
+        type="button"
+        disabled={isEmpty(emailForCheckout)}
+      >
+        Become a Member
+      </button>
+      {pppCouponAvailable && pppCouponEligible && (
+        <div className="mt-4 pb-5 max-w-screen-md mx-auto">
+          <SmallParityCouponMessage
+            coupon={parityCoupon as Coupon}
+            countryName={countryName as string}
+            onApply={onApplyParityCoupon}
+            onDismiss={onDismissParityCoupon}
+            isPPP={pppCouponIsApplied}
+          />
```

Comment:
I created a slimmed down version of the Parity Coupon Message, relative to the one that displays on the pricing page.